namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -417,6 +424,48 @@ class HttpControlService(storage: DtabStore, delegate: Ns => NameInterpreter, na
     }
   }
 
+  private[this] def flattenTree(tree: NameTree[Name.Bound], rng: Rng = Rng.threadLocal): Option[Path] = {
+    tree match {
+      case NameTree.Neg | NameTree.Fail | NameTree.Empty | NameTree.Alt() => None
+      case NameTree.Leaf(bound) => bound.id match {
+        case p: Path => Some(p)
+        case _ => None
+      }
+      case NameTree.Union(weightedTrees@_*) =>
+        val (weights, trees) = weightedTrees.unzip { case NameTree.Weighted(w, t) => (w, t) }
+        val drv = Drv.fromWeights(weights)
+        val randomTree = trees(drv(rng))
+        flattenTree(randomTree, rng)
+      case NameTree.Alt(alts@_*) =>
+        flattenTree(alts.head) match {
+          case p: Some[Path] => p
+          case None => flattenTree(NameTree.Alt(alts.tail: _*), rng)
+        }
+    }
+  }
+
+  private[this] def handleGetResolve(ns: String, path: Path, req: Request): Future[Response] = {
+    val extraDtab = req.params.get("dtab")
+
+    val activity = getBind(ns, path, extraDtab).flatMap { tree =>
+      flattenTree(tree) match {
+        case p: Some[Path] => addrCache.get((ns, p.get))
+        case None => Activity.value(Addr.Neg)
+      }
+    }
+
+    if (isStreaming(req)) {
+      streamingResp(activity.values) { addr => Buf.Utf8(addr.toString + "\n") }
```

Comment:
For now I changed it to call to `DelegateApiHandler.Codec` but I see #575 - once it's merged (assuming it would be merged before this one) I'd be happy to change it to updated `renderAddr` that makes the same thing.